src/rgw/rgw_op.cc

```diff
@@ -2519,15 +2519,24 @@ void RGWPutObj::execute()
   }
 
   do {
-    bufferlist data;
-    len = get_data(data);
+    bufferlist data_in;
+    len = get_data(data_in);
     if (len < 0) {
       op_ret = len;
       goto done;
     }
     if (!len)
       break;
 
+    bufferlist data;
```

Comment:
This induces buffer sharing.  Would we get the same result if data were a pointer to data_in or s->aws_auth->bl (saving the atomic ops to attach each buffer::ptr)?